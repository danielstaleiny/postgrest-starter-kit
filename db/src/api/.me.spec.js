const ROOT = process.cwd()
const { get, post, put, patch, del } = require(`${ROOT}/util/fetch.js`)(
  'http://localhost:3000/'
)

describe('/rpc/me', () => {
  test('success 200 POST', async () => {
    const res = await post('rpc/me')({
      withRole: { id: 1, role: 'webuser' },
    })
    expect(res).toMatchSnapshot()
  })

  test('success 200 GET', async () => {
    const res = await get('rpc/me')({
      withRole: { id: 1, role: 'webuser' },
    })
    expect(res).toMatchSnapshot()
  })

  test('fail 401', async () => {
    const res = await post('rpc/me')({})
    expect(res).toMatchSnapshot()
  })
})
