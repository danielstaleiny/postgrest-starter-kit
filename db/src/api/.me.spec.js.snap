// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/rpc/me fail 401 1`] = `
Object {
  "body": Object {
    "code": "42501",
    "details": null,
    "hint": null,
    "message": "permission denied for function me",
  },
  "status": 401,
  "statusText": "Unauthorized",
}
`;

exports[`/rpc/me success 200 GET 1`] = `
Object {
  "body": Object {
    "email": "alice@email.com",
    "id": 1,
    "name": "alice",
    "role": "webuser",
  },
  "status": 200,
  "statusText": "OK",
}
`;

exports[`/rpc/me success 200 POST 1`] = `
Object {
  "body": Object {
    "email": "alice@email.com",
    "id": 1,
    "name": "alice",
    "role": "webuser",
  },
  "status": 200,
  "statusText": "OK",
}
`;
